fix: Hide spammy & debug draw NetworkTransform behind define [MTT-1262]

JesseOlmer opened the pull request:
- Moves debug draw & local-write warnings in NT behind a debug define
- Fixes warning message format
- Fixes teleport exception message

Future changes not addressed here:
- Debug draw should be via gizmos, not the current implementation
- Spammy log messages shouldn't be so spammy, and shouldn't be so expensive to detect

Commit:
chore: Hide spammy & debug draw NetworkTransform behind define

- Moves debug draw & local-write warnings in NT behind a debug define
- Fixes warning message format
- Fixes teleport exception message

Future changes not addressed here:
- Debug draw should be via gizmos, not the current implementation
- Spammy log messages shouldn't be so spammy, and shouldn't be so expensive to detect

SamuelBellomo commented on com.unity.netcode.gameobjects/Components/NetworkTransform.cs:
```diff
@@ -840,25 +838,28 @@ protected virtual void Update()
 
                 if (!CanCommitToTransform)
                 {
+#if NGO_TRANSFORM_DEBUG
```
any reason why we're not using NetworkManager.LogLevel == LogLevel.Developer for this?

SamuelBellomo replied:
This would make sure we only have one place to enable/disable debug "things" instead of having to manage both log level + define

SamuelBellomo replied:
it just feels like if we have this, no one will use it and it might just be better to remove this code block altogether.
The thinking behind the warning was to help users debug why they were seeing their transform not moving client side when they expected it to. This was a sort of "educate through the console" type of thing, to educate on server authority and why you shouldn't try to move things client side.
if we hide this behind a define, we might as well just remove this warning, it'd simplify our code.

JesseOlmer replied:
- The warning is broken in multiple ways
- The warning creates garbage even before you call debug.log
- The warning does not use NetworkLog
- The warning fires in cases where it shouldn't (e.g. a changed rotY when only rotX is synced)
- This code also renders to screen with debug draws, guarded by a log filter - which is not appropriate.

As you can see there are a significant number of issues with this code and fixing it completely would require a substantial effort that I'm not comfortable taking at this stage. However, this log has proved useful for our own internal troubleshooting so I didn't want to delete it altogether. Hence, a debug define.

Commit:
fix: Test log verification workaround.

We have a test which verifies log output. Updated the test to match same parameters (that is, define guard and developer log level guard).

These tests need to be properly refactored post-1.0

Commit:
Merge branch 'develop' into fix/networktransform-spam

Commit:
test: guard regex using in test file with NGO_TRANSFORM_DEBUG

Commit:
Merge branch 'develop' into fix/networktransform-spam

mattwalsh-unity committed:
Merge branch 'develop' into fix/networktransform-spam

mattwalsh-unity commented:
I incorporated your changes in 1286 after unwittingly duplicating the work.  This PR won't be needed when 1286 goes in
